js/main.js

```diff
@@ -18,83 +21,143 @@ function updatePhoto(event) {
-    if ($viewElements[i].getAttribute('data-view') !== 'entries') {
-      $viewElements[i].classList.add('hidden');
-    } else $viewElements[i].classList.remove('hidden');
+window.addEventListener('DOMContentLoaded', function (event) {
+  dataEntryId = data.entries.length;
+  for (i = 0; i < data.entries.length; i++) {
+    var entryValues = renderEntry(data.entries[i]);
+    $ul.appendChild(entryValues);
+    dataEntryId--;
   }
-  swapViews('entries');
-  i = 0;
-  var $entryValues = renderEntry(data.entries[i]);
-  $ul.prepend($entryValues);
-  $entriesEmpty.classList.add('hidden');
-}
+  hideEmptyEntriesPage();
+  swapViews(data.view);
+});
 
 function renderEntry(newEntry) {
   var $li = document.createElement('li');
   $li.className = 'row padding-bottom';
+  $li.setAttribute('data-entry-id', dataEntryId);
 
   var $divImage = document.createElement('div');
   $divImage.className = 'column-full column-half image-container margin-bottom';
 
-  var $divText = document.createElement('div');
-  $divText.className = 'column-full column-half padding-left';
-
   var $entryImage = document.createElement('img');
   $entryImage.className = 'entry-image';
   $entryImage.setAttribute('src', data.entries[i].photoUrl);
 
+  var $divText = document.createElement('div');
+  $divText.className = 'column-full column-half padding-left';
+
+  var $entryHeading = document.createElement('div');
+  $entryHeading.className = 'row entry-heading';
+
   var $entryTitle = document.createElement('h1');
-  $entryTitle.className = 'entry-title';
   $entryTitle.textContent = data.entries[i].title;
 
+  var $divEditIcon = document.createElement('div');
+  $divEditIcon.className = 'edit-icon';
+
+  var $editIcon = document.createElement('i');
+  $editIcon.className = 'fas fa-pen';
+
   var $entryNote = document.createElement('p');
   $entryNote.className = 'entry-note';
   $entryNote.textContent = data.entries[i].notes;
 
   $li.appendChild($divImage);
   $li.appendChild($divText);
   $divImage.appendChild($entryImage);
-  $divText.appendChild($entryTitle);
+  $divText.appendChild($entryHeading);
   $divText.appendChild($entryNote);
+  $entryHeading.appendChild($entryTitle);
+  $entryHeading.appendChild($divEditIcon);
+  $divEditIcon.appendChild($editIcon);
 
   return $li;
 }
 
-window.addEventListener('DOMContentLoaded', function (event) {
-  for (i = 0; i < data.entries.length; i++) {
-    var $entryValues = renderEntry(data.entries[i]);
-    $ul.appendChild($entryValues);
-  }
+function createNewEntry() {
+  var newEntry = {
+    entryId: data.nextEntryId,
+    photoUrl: $photoUrl.value,
+    title: $title.value,
+    notes: $notes.value
+  };
 
-  if (data.entries.length > 0) {
-    $entriesEmpty.classList.add('hidden');
-  } else $entriesEmpty.classList.remove('hidden');
+  data.nextEntryId++;
+  data.entries.unshift(newEntry);
+}
 
-  swapViews(data.view);
-});
+function addEntryToPage() {
+  i = 0;
+  var entryValues = renderEntry(data.entries[i]);
```

Comment:
I think you might be able to do away with this global `i` variable. In both cases I see you using it, you are assigning a value to it on the line above. Instead, you can just use that value directly. for instance here, you can just pass `data.entries[0]` as the argument to `renderEntry`